libs/blocks/table-metadata/table-metadata.js

```diff
@@ -0,0 +1,138 @@
+import { getMetadata, handleStyle } from '../section-metadata/section-metadata.js';
+
+const getIndexedValues = (text) => text.split('\n').map((value) => value.split(/,(.*)/s).map((v) => v.trim()));
+
+function handleHighlightMerch(text, table) {
+  if (!text) return;
+  const highLights = getIndexedValues(text);
+  highLights.forEach((hl) => {
+    const hlCol = hl[0];
+    const hlText = hl[1];
+    if (hlText) {
+      const highLightDiv = document.createElement('div');
+      highLightDiv.textContent = hlText;
+      table.querySelector(`.col-${hlCol}`).classList.add('col-highlight');
+      table.querySelector(`.col-${hlCol}`).prepend(highLightDiv);
+    }
+  });
+}
+
+function handleSectionHead(text, table) {
+  if (!text) return;
+  const sectionsHeads = text.split('\n');
+  sectionsHeads.forEach(sh => {
+    const sectionHead = table.querySelector(`.row-${sh.trim()}`);
+    sectionHead.classList.add('sectionHead');
+  });
+}
+
+const handleColumnColorMerch = (text, table, columnType) => {
+  if (!text) return;
+  const colors = getIndexedValues(text);
+  colors.forEach((color) => {
+    const colorIndex = color[0];
+    const colorValue = color[1]?.replace('white', 'light');
+    table.querySelector(`.col-${colorIndex}.col-${columnType}`)?.classList.add(colorValue);
+  });
+};
+
+const handleColumnBgColorMerch = (text, table, columnType) => {
+  if (!text) return;
+  const bgColors = getIndexedValues(text);
+  bgColors.forEach((color) => {
+    const bgColorIndex = color[0];
+    const bgColorValue = color[1];
+    const col = table.querySelector(`.col-${bgColorIndex}.col-${columnType}`);
+    if (col) col.style.background = bgColorValue;
+  });
+};
+
+function handleCompare(text, table) {
+  if (!(text || table)) return;
+  const comparisonGroup  = text.split('\n');
+  comparisonGroup.forEach((comp, i) => {
+    const col = comp.trim().split(' ')[1];
+    const comparable = table.querySelector(`.col-${col}`);
+    comparable.classList.add(`comp_${i + 1}`);
+  });
+}
+
+function handleCollapse(collapsRows, expandDefault, table) {
+  if (collapsRows) {
+    const collapseGroups = collapsRows.split('\n');
+    collapseGroups.forEach((collapseGroup) => {
+      const rowsGroup = collapseGroup.trim().split(' ');
+      rowsGroup.forEach((group) => {
+        if (group.includes('-')) {
+          const [rowStartStr, rowEndStr] = group.trim().split('-');
+          const rowStart = Number(rowStartStr) + 1;
+          const rowEnd = Number(rowEndStr);
+          if (rowStart && rowStart) {
```

Comment:
I assume should be `rowStart && rowEnd`.  Also can early return to reduce nesting